tq/transfer_queue.go

```diff
@@ -530,10 +559,10 @@ func (q *TransferQueue) enqueueAndCollectRetriesFor(batch batch) (batch, error)
 			// retried, they will be marked as failed.
 			for _, t := range batch {
 				if q.canRetryObject(t.Oid, err) {
-					q.rc.Increment(t.Oid)
-
-					next = append(next, t)
+					enqueueRetry(t, err)
 				} else if readyTime, canRetry := q.canRetryObjectLater(t.Oid, err); canRetry {
+					// Do not use enqueueRetry() here because retries requested
+					// by a server do not count against the normal retry limit
```

Comment:
I don't know if this is actually the desired behavior: `rc.Increment` is not called, but it's unclear if that was intentional or an oversight.

Reply:
I definitely think we should count them against the limit.